train.py

```diff
@@ -152,6 +154,12 @@ def train(
 if __name__ == "__main__":
     """Main function to run the training of the model."""
 
+    # Set the GPU to allocate only used memory at runtime.
```

Comment:
Distributed training can be done, in torch, it's pretty simple, I don't know about TensorFlow.